README.md

```diff
@@ -162,37 +163,396 @@ fun main(args: Array<String>) {
+        Cairo0ContractDefinition contractDefinition = new Cairo0ContractDefinition(contractCode);
+        // Class hash is calculated using the tools you used for compilation (only for Cairo v0 contracts)
+        Felt classHash = Felt.fromHex("0x1a3b2c");
+        Felt nonce = account.getNonce().send();
+
+        // Estimate fee for declaring a contract
+        DeclareTransactionV1Payload declareTransactionPayloadForFeeEstimate = account.signDeclare(contractDefinition, classHash,new ExecutionParams(nonce, new Felt(1000000000000000L)), false);
+        Request<List<EstimateFeeResponse>> feeEstimateRequest = provider.getEstimateFee(List.of(declareTransactionPayloadForFeeEstimate));
+        Felt feeEstimate = feeEstimateRequest.send().get(0).getOverallFee();
+        // Make sure to prefund the account with enough funds to cover the fee for declare transaction
+
+        // Declare a contract
+        ExecutionParams params = new ExecutionParams(nonce, new Felt(feeEstimate.getValue().multiply(BigInteger.TEN)));
+        DeclareTransactionV1Payload declareTransactionPayload = account.signDeclare(contractDefinition, classHash, params, false);
+
+        Request<DeclareResponse> request = provider.declareContract(declareTransactionPayload);
+        DeclareResponse response = request.send();
+        Felt hash = response.getTransactionHash(); // Hash of the transaction that declares a contract
+    }
+}
+```
+or in Kotlin
+
+```kotlin
+fun main(args: Array<String>) {
+    // Create a provider for interacting with StarkNet
+    val provider = JsonRpcProvider("https://example-node-url.com/rpc", StarknetChainId.TESTNET)
+
+    // Set up an account
+    val privateKey = Felt.fromHex("0x123")
+    val accountAddress = Felt.fromHex("0x1236789")
+    // ⚠️ WARNING ⚠️ Both the account address and private key are for demonstration purposes only.
+    val account = StandardAccount(accountAddress, privateKey, provider)
+
+    // Import a compiled contract
+    val contractCode = Path.of("contract.json").readText()
+    val contractDefinition = Cairo0ContractDefinition(contractCode)
+    // Class hash is calculated using the tools you used for compilation (only for Cairo v0 contracts)
+    val classHash = Felt.fromHex("0x1a3b2c")
+    val nonce = account.getNonce().send()
+
+    // Estimate fee for declaring a contract
+    val declareTransactionPayloadForFeeEstimate = account.signDeclare(
+        contractDefinition = contractDefinition,
+        classHash = classHash,
+        params = ExecutionParams(nonce, Felt(1000000000000000L)),
+        forFeeEstimate = true,
+    )
+    val feeEstimateRequest = provider.getEstimateFee(listOf(declareTransactionPayloadForFeeEstimate))
+    val feeEstimate = feeEstimateRequest.send().first().overallFee
+    // Make sure to prefund the account with enough funds to cover the fee for declare transaction
+
+    // Declare a contract
+    val params = ExecutionParams(nonce, Felt(feeEstimate.value.multiply(BigInteger.TEN)))
```

Comment:
Even though `nonce` speaks for itself, I'd use named arguments here and in other similar classes, especially because these are code examples for documentation